dev/build-tracker/slack.go

```diff
@@ -62,15 +75,34 @@ func NewNotificationClient(logger log.Logger, slackToken, githubToken, channel s
 	teamResolver := team.NewTeammateResolver(githubClient, slackClient)
 
 	return &NotificationClient{
-		logger:  logger.Scoped("notificationClient", "client which interacts with Slack and Github to send notifications"),
-		slack:   *slackClient,
-		team:    teamResolver,
-		channel: channel,
+		logger:              logger.Scoped("notificationClient", "client which interacts with Slack and Github to send notifications"),
+		slack:               *slackClient,
+		team:                teamResolver,
+		commitTeammateCache: map[string]cacheItem[*team.Teammate]{},
```

Comment:
```suggestion
		commitTeammateCache: map[string]*cacheItem[*team.Teammate]{},
```

This would simplify and remove a copy.